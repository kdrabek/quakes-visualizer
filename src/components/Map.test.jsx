import Immutable from 'immutable';
import { shallow } from 'enzyme';
import EnhancedMap, { Map } from './Map';

const coords = Immutable.fromJS({ lat: 12.34, lng: 34.21 });

describe('EnhancedMap component', () => {
  it('should match snapshot when coords not passed', () => {
    const props = {};
    const wrapper = shallow(<EnhancedMap {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot when coords passed', () => {
    const props = { mapCoords: coords };
    const wrapper = shallow(<EnhancedMap {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Map component', () => {
  it('should match snapshot', () => {
    const props = {
      center: coords.toJS(),
      loadingElement: <div />,
      googleMapURL: 'https://maps.url.com',
    };
    const wrapper = shallow(<Map {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
