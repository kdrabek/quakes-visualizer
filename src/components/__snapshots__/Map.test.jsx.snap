// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnhancedMap component should match snapshot when coords not passed 1`] = `
<withScriptjs(withGoogleMap(Component))
  center={
    Object {
      "lat": 51.28,
      "lng": 0,
    }
  }
  containerElement={
    <div
      className="wrapper"
    />
  }
  defaultZoom={5}
  googleMapURL="https://maps.googleapis.com/maps/api/js?v=3.exp&libraries=geometry,drawing,places"
  isMarkerShown={true}
  loadingElement={
    <div
      className="wrapper"
    />
  }
  mapElement={
    <div
      className="wrapper"
    />
  }
/>
`;

exports[`EnhancedMap component should match snapshot when coords passed 1`] = `
<withScriptjs(withGoogleMap(Component))
  center={
    Object {
      "lat": 12.34,
      "lng": 34.21,
    }
  }
  containerElement={
    <div
      className="wrapper"
    />
  }
  defaultZoom={5}
  googleMapURL="https://maps.googleapis.com/maps/api/js?v=3.exp&libraries=geometry,drawing,places"
  isMarkerShown={true}
  loadingElement={
    <div
      className="wrapper"
    />
  }
  mapElement={
    <div
      className="wrapper"
    />
  }
/>
`;

exports[`Map component should match snapshot 1`] = `<div />`;
